network/src/definitions.ts

```diff
@@ -0,0 +1,28 @@
+import { PluginListenerHandle } from '@capacitor/core';
+
+export interface NetworkPlugin {
+  /**
+   * Query the current network status
+   */
+  getStatus(): Promise<NetworkStatus>;
+
+  /**
+   * Listen for network status change events
+   */
+  addListener(
+    eventName: 'networkStatusChange',
+    listenerFunc: (status: NetworkStatus) => void,
+  ): PluginListenerHandle;
+
+  /**
+   * Remove all native listeners for this plugin
+   */
+  removeAllListeners(): void;
+}
+
+export interface NetworkStatus {
+  connected: boolean;
+  connectionType: 'wifi' | 'cellular' | 'none' | 'unknown';
+}
+
+export type NetworkStatusChangeCallback = (status: NetworkStatus) => void;
```

Comment:
Please fully document this file (see [Storage plugin](https://github.com/ionic-team/capacitor-plugins/pull/15/files#diff-5f68b89cfcbd395c3f9ecef3bbdf9852) for an example). Feel free to rewrite the sentences, too. 

The JSDoc you write will be available for devs when they use these APIs in VS Code. Additionally, we can use it to autogenerate documentation for this plugin.